### PR TITLE
Improve error message for CLI param validation

### LINKED ISSUE #9846: Confusing CLI string flag validation error message
Steps to reproduce:
Deleting `~/.aws/config` and `~/.aws/credentials`
Run `./node_modules/.bin/serverless config credentials --provider aws --key $AWS_KEY_ID --secret $AWS_SECRET_ACCESS_KEY --profile serverless --overwrite`

Actual Behavior:
```
  ServerlessError: Unexpected boolean "--key"
      at module.exports (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/parse-args.js:59:15)
      at /Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/resolve-input.js:55:29
      at memoized (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/memoizee/lib/configure-map.js:80:41)
      at module.exports (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/handle-error.js:176:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Expected Behavior:
No error, and the configuration to be written to a new `~/.aws/credentials` file in the specified profile ("serverless" was the profile in this case).

This also happens (apparently intermittently) on a clean github actions runner for example https://github.com/activescott/serverless-aws-static-file-handler/pull/96/checks?check_run_id=3215635986#step:5:74

NOTE: I do not think that the serverless.yml is relevant here, but I have provided it (also [on github](https://github.com/activescott/serverless-aws-static-file-handler/blob/b34054baa85c6e3f4c9fb359ddf70e486e6504ac/examples/basic/serverless.yml))

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: static-file-handler-demo-proxy

plugins:
  - serverless-aws-static-file-handler/plugins/BinaryMediaTypes

custom:
  apiGateway:
    binaryMediaTypes:
      - image/png
      - image/jpeg
      - "*/*" # Using */* doesn't have side effects that I can tell. The HTML response works fine.

provider:
  name: aws
  runtime: nodejs14.x
  lambdaHashingVersion: 20201221

functions:
  html:
    handler: handler.html
    events:
      - http:
          path: /
          method: get
  png:
    handler: handler.png
    events:
      - http:
          path: png
          method: get
  jpg:
    handler: handler.jpg
    events:
      - http:
          path: jpg
          method: get
  binary:
    handler: handler.binary
    events:
      - http:
          path: /binary/{pathvar+}
          method: get

```
</details>

<details>
<summary><b><code>SLS_DEBUG=* ./node_modules/.bin/serverless config credentials --provider aws --key $AWS_KEY_ID --secret $AWS_SECRET_ACCESS_KEY --profile serverless --overwrite</code></b></summary>

```
 Serverless Error ----------------------------------------
 
  ServerlessError: Unexpected boolean "--key"
      at module.exports (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/parse-args.js:59:15)
      at /Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/resolve-input.js:55:29
      at memoized (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/memoizee/lib/configure-map.js:80:41)
      at /Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/scripts/serverless.js:111:69
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              14.17.3
     Framework Version:         2.54.0 (local)
     Plugin Version:            5.4.3
     SDK Version:               4.2.6
     Components Version:        3.15.0
 
 
 Serverless Error ----------------------------------------
 
  ServerlessError: Unexpected boolean "--key"
      at module.exports (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/parse-args.js:59:15)
      at /Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/resolve-input.js:55:29
      at memoized (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/memoizee/lib/configure-map.js:80:41)
      at module.exports (/Users/scott/src/activescott/serverless-aws-static-file-handler/examples/basic/node_modules/serverless/lib/cli/handle-error.js:178:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              14.17.3
     Framework Version:         2.54.0 (local)
     Plugin Version:            5.4.3
     SDK Version:               4.2.6
     Components Version:        3.15.0

```
</details>

<b>Installed version</b>

```
Framework Core: 2.54.0 (local)
Plugin: 5.4.3
SDK: 4.2.6
Components: 3.15.0
```

### PR DESCRIPTION
**Pull request by mnapoli**
Closes #9846

### CODE REVIEW

File: lib/cli/parse-args.js [hunk 1]
```diff
@@ -59,7 +59,10 @@ module.exports = (
     if (string.has(paramName) || multiple.has(paramName)) {
       if (isBoolean) {
         if (isHelpRequest) return result;
-        throw new ServerlessError(`Unexpected boolean "--${paramName}"`, 'MISSING_CLI_PARAM_VALUE');
+        throw new ServerlessError(
```
Comment (pgrzesik) [inline lib/cli/parse-args.js:62]: I'm honestly not sure about it. How are we sure that there's actually a missing value for a CLI parameter and not an unexpected boolean flag being passed?

Comment (mnapoli) [reply lib/cli/parse-args.js:62]: @pgrzesik I don't know if that's the case here, but maybe it is.

I did a quick check and I get a different error message:

image

Comment (pgrzesik) [reply lib/cli/parse-args.js:62]: oh nice 👍

**Review by medikoo — APPROVED** [review]
Yes, that looks great! It also doesn't need test change and we just change the error message

**Pull request merged**